go/vt/vttablet/tabletserver/schema/load_table.go

```diff
@@ -117,20 +102,49 @@ func loadMessageInfo(ta *Table, comment string) error {
 
 	ta.MessageInfo.MaxBackoff, _ = getDuration(keyvals, "vt_max_backoff")
 
+	// configure the columns that will be loaded into memory and streamed to subscribers
+	hiddenCols := map[string]struct{}{
+		"priority":   {},
+		"time_next":  {},
+		"epoch":      {},
+		"time_acked": {},
+	}
+
+	requiredCols := []string{
+		"id",
+		"priority",
+		"time_next",
+		"epoch",
+		"time_acked",
+	}
+
+	// make sure required columns are present
 	for _, col := range requiredCols {
 		num := ta.FindColumn(sqlparser.NewColIdent(col))
 		if num == -1 {
 			return fmt.Errorf("%s missing from message table: %s", col, ta.Name.String())
 		}
 	}
 
-	// Load user-defined columns. Any "unrecognized" column is user-defined.
-	for _, field := range ta.Fields {
-		if _, ok := hiddenCols[strings.ToLower(field.Name)]; ok {
-			continue
+	allowedStreamingCols, _ := getCols(keyvals, "vt_stream_cols")
+
+	// make sure that all the specified columns are present
```

Comment:
This is where `allowed` feels misleading as you're saying that they are required. If they were allowed then I would expect that them not being present would be fine.

Reply:
renamed to `specifiedCols`